### PR TITLE
[DOC] Fix description in CapStyle example

### PR DESCRIPTION
**Pull request by otaithleigh**
## PR summary

The current [CapStyle example](https://matplotlib.org/devdocs/gallery/lines_bars_and_markers/capstyle.html) partially uses text from the [JoinStyle example](https://matplotlib.org/devdocs/gallery/lines_bars_and_markers/joinstyle.html). I've updated the text with the appropriate description.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
Thank you for catching and fixing this @otaithleigh !

We will wait for the docs CI to pass before merging (because it _should_ build but I do not trust my dead-reckoning of rst) before merging.

**Comment by timhoffm** [comment]
Thank, and congratulations on your first contribution to matplotlib! 🎉

**Pull request merged**
